"""2. Создайте собственный класс-исключение, обрабатывающий ситуацию деления на нуль.
Проверьте его работу на данных, вводимых пользователем. При вводе пользователем нуля в
качестве делителя программа должна корректно обработать эту ситуацию и не завершиться с ошибкой."""

class MyError(Exception):


    def __init__(self, txt):
        self.txt = txt


a = int(input('введите число а: '))
b = int(input('введите число b: '))

try:
    if b == 0:
        raise MyError('вы попытались делить на 0')
    result = a/b
except MyError as arr:
    print(f'{arr}')
